[cyrolite] iP

cyrolite opened the pull request:
# **Scooby!**

> "The Key Is In Not Spending Time, But In Investing It." - Stephen R. Covey ([source](https://www.forbes.com/sites/katecooper/2022/07/03/meetings-the-key-is-in-not-spending-time-but-in-investing-it--stephen-r-covey/))

Feeling overwhelmed with deadlines, assignments, and commitments? We get it—life is hectic, and keeping track of everything can be a nightmare. That is why Scooby is here to help!

## Introducing Scooby!

Your **personalized** task manager, designed to help you stay on top of your work **effortlessly**—so you can focus on _what truly matters!_

Scooby is:

- text-based
- easy to learn
- user-friendly
- ~FAST~ SUPER FAST to use 👍 

## How to access Scooby?

1. Download the files [here](https://github.com/nus-cs2103-AY2425S2/ip/pull/108).
2. Go to Scooby.java and run the program:
```    
public static void main(String[] args) {
        Scooby sc = new Scooby();
        sc.run();
}
```
3. Add / delete tasks on your command!

It is that **SIMPLE**!

Features of Scooby:

- [X]  Add or Delete your tasks
- [X] List all of your tasks
- [X] Find any tasks to complete
- [ ] Managing deadlines (coming soon)
- [ ] Reminders (coming soon)

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

cyrolite committed:
Rename the name from Duke to Scooby, and add greeting and exit dialogues

cyrolite committed:
Change exit_dialogue() to exitDialogue

cyrolite committed:
Add new class Ui for interaction support, change Scooby class

cyrolite committed:
Add new TaskList, Ui class, and updated Scooby class.

cyrolite committed:
Added Parser and Task classes, edited TaskList, Ui and Scooby class

cyrolite committed:
Edited TaskList class, added Deadline, Event, ToDo classes

cyrolite committed:
Add new Input, Expected test cases, edit runtest.bat

cyrolite committed:
Add exceptions, update test cases, TaskList and Parser

cyrolite committed:
Minor changes to error messages

cyrolite committed:
Delete Task added to the code

cyrolite committed:
Add more test cases

cyrolite committed:
Amend Parser.java, TaskList.java, add tasks.txt

cyrolite committed:
Merge branch 'branch-Level-7'

cyrolite committed:
Include LocalDateTime in Deadline.java and Event.java

cyrolite committed:
Edit Event.java, Task.java, TaskList.java and Deadline.java

cyrolite committed:
Edit Javadoc comments, added new branch

cyrolite committed:
Improve fomatting for Event.java

cyrolite committed:
Merge branch 'branch-Level-8'

cyrolite committed:
Initialise OOP principles in Scooby.java

cyrolite committed:
Add Line.java for printing of Lines

cyrolite committed:
Removed unnecessary functions in Scooby.java

cyrolite committed:
Merge branch 'A-MoreOOP'

cyrolite committed:
Divide classes into packages

cyrolite committed:
Merge branch 'A-Packages'

cyrolite committed:
Edit java files to include packages

cyrolite committed:
Merge branch 'add-gradle-support'

cyrolite committed:
Add build.gradle

cyrolite committed:
Amend packages

cyrolite committed:
Add unit tests

cyrolite committed:
Merge branch 'A-JUnit'

cyrolite committed:
Added jar file

cyrolite committed:
Merge branch 'A-Jar'

cyrolite committed:
Improved documentation

cyrolite committed:
Merge branch 'A-JavaDoc'

cyrolite committed:
Format imports in order

cyrolite committed:
Merge branch 'A-CodingStandard'

cyrolite committed:
Add find feature

cyrolite committed:
Ignore casing in find feature

cyrolite committed:
Merge branch 'branch-Level-9'

cyrolite committed:
Add image support

cyrolite committed:
Add fxml files

cyrolite committed:
Add basic GUI

cyrolite committed:
Debug files

cyrolite committed:
Fix "bye" command

cyrolite committed:
Fix more bugs with UI

cyrolite committed:
Merge branch 'branch-Level-10'

cyrolite committed:
Improve UI

cyrolite committed:
Increase text bubble size

cyrolite committed:
Add background

cyrolite committed:
Add text prompt to GUI

cyrolite committed:
Format UI

chen-xiangrui reviewed (commented):
LGTM! I like how the structure of your iP is very neat and tidy. The coding standards are also closely adhered to!

chen-xiangrui commented on src/main/java/scooby/tasks/TaskList.java:
```diff
@@ -0,0 +1,259 @@
+import java.io.IOException;
+import java.io.File;
+import java.io.BufferedWriter;
+import java.io.FileWriter;
```
You could do some organising of the imported libraries and group those from the same ones (eg. java.io.*) together, with a blank line separating them from java.util.ArrayList

chen-xiangrui commented on src/main/java/scooby/tasks/Task.java:
```diff
@@ -0,0 +1,60 @@
+    }
+
+    /**
+     * Returns the raw representation of the String to be read from the txt file
```
The JavaDoc syntax is simple and easy to understand

chen-xiangrui commented on src/test/java/scooby/ParserTest.java:
```diff
@@ -0,0 +1,69 @@
+    }
+
+    @Test
+    void testParseCommand_AddTask_EmptyDescription() {
```
Useful test case to test the edge cases

chen-xiangrui reviewed (commented):
LGTM! I like how the structure of your iP is very neat and tidy. The coding standards are also closely adhered to!

cyrolite committed:
Amend build.gradle

cyrolite committed:
Add Storage.java

cyrolite committed:
Ignore tasks.txt

cyrolite committed:
Remove redundant method

cyrolite committed:
Format Parser.java

cyrolite committed:
Add assert in TaskList.java

cyrolite committed:
Add assert in Storage.java

cyrolite committed:
Update assert in Parser.java

cyrolite committed:
Revert "Update assert in Parser.java"

This reverts commit e74b9edab40678f87bc081597dade55a4026fa88.

cyrolite committed:
Update assert in Tasklist.java

cyrolite committed:
Update assert in Storage.java

cyrolite committed:
Add assert in Parser.java

cyrolite committed:
Merge pull request #1 from cyrolite/A-Assertions

A-Assertions

cyrolite committed:
Refactor some classes for improved consistency and maintainability

There was a lack of consistency in the code. As such, several improvements were made to enhance readability, maintainability, and structure, which includes:
1. Refactored 'Storage.java' and 'Parser.java' by replacing repetitive if-else conditions with a switch-case.
2. Extracted redundant file handling logic into separate helper methods.
3. Removed unnecessary methods and commented-out codes.

All of the above was implemented to improve the readability of the code, and ensures that the Code Quality is up to standard.

cyrolite committed:
Refactor code for Task.java

cyrolite committed:
Add Streams to TaskList.java

cyrolite committed:
Merge branch 'branch-A-Streams' into heads/A-CodeQuality

cyrolite committed:
Merge pull request #2 from cyrolite/heads/A-CodeQuality

A-CodeQuality

cyrolite committed:
Merge pull request #3 from cyrolite/branch-A-Streams

Branch-A-streams

cyrolite committed:
Add detect duplicates

cyrolite committed:
Add update commands

cyrolite committed:
Merge pull request #4 from cyrolite/heads/BCD-Extension

BCD-extension

cyrolite committed:
Add help feature

cyrolite committed:
Merge pull request #5 from cyrolite/branch-Help

Add help feature

cyrolite committed:
Remove small print debug statements

cyrolite committed:
Edit javadoc comments

cyrolite committed:
Add user guide

cyrolite committed:
Update README.md

cyrolite committed:
Update README.md

cyrolite committed:
Update README.md

cyrolite committed:
Refactor code to get title in the Ui

cyrolite committed:
Merge pull request #7 from cyrolite/Refactoring

Refactor code to get title in the Ui
